sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/OperatorStateMetadata.scala

```diff
@@ -468,7 +468,13 @@ class OperatorStateMetadataV2FileManager(
     // find the batchId of the earliest schema file we need to keep
     val earliestBatchToKeep = latestMetadata match {
       case Some(OperatorStateMetadataV2(_, stateStoreInfo, _)) =>
-        val schemaFilePath = stateStoreInfo.head.stateSchemaFilePath
+        // Currently, this purging logic only handles operators with a single state store
+        // instance, which is why we choose the head of this array.
+        val ssInfo = stateStoreInfo.head
+        // The schema files in the OperatorStateMetadata are sorted
+        // by earliest to latest. The head of the schemaFileList is
+        // the schema file written for batch 0.
```

Comment:
nit: not batch 0 but the earliest batch if we are ever evicting them? If we aren't evicting, it's simply always 0 and we don't need to do this.